pkg/transform/oauth/basicauth.go

```diff
@@ -16,21 +16,23 @@ type IdentityProviderBasicAuth struct {
 	BasicAuth              BasicAuth `yaml:"basicAuth"`
 }
 
+// BasicAuth provider specific data
 // BasicAuth provider specific data
 type BasicAuth struct {
-	URL           string        `yaml:"url"`
-	CA            CA            `yaml:"ca"`
-	TLSClientCert TLSClientCert `yaml:"tlsClientCert"`
-	TLSClientKey  TLSClientKey  `yaml:"tlsClientKey"`
+	URL           string         `yaml:"url"`
+	CA            *CA            `yaml:"ca,omitempty"`
```

Comment:
Are we safe from nil pointer dereference's etc. with these. I'm fine with it as long as we're not as risk of stuff unexpectedly blowing up.

Reply:
I added tests where this values are empty, should be safe. Is there any other way to omit empty values when working with structures?

Reply:
If there are tests to cover it I'm good. Thanks!